lib/validation.js

```diff
@@ -114,11 +114,9 @@ function validate (context, request) {
 }
 
 function wrapValidationError (result, dataVar) {
-  if (result instanceof Error) {
-    return result
-  }
-  var error = new Error(schemaErrorsText(result, dataVar))
+  const error = (result instanceof Error) ? result : new Error(schemaErrorsText(result, dataVar))
   error.validation = result
```

Comment:
I don't think we should do this assignment if `result instanceof Error`.

Reply:
You're probably right, I wasn't sure...